src/partition.rs

```diff
@@ -355,32 +355,32 @@ use plane::*;
 use predict::*;
 
 impl PredictionMode {
-  pub fn predict_intra<'a>(self, dst: &'a mut PlaneMutSlice<'a>, tx_size: TxSize) {
+  pub fn predict_intra<'a>(self, dst: &'a mut PlaneMutSlice<'a>, tx_size: TxSize, bit_depth: usize) {
     assert!(self.is_intra());
 
     match tx_size {
-      TxSize::TX_4X4 => self.predict_intra_inner::<Block4x4>(dst),
-      TxSize::TX_8X8 => self.predict_intra_inner::<Block8x8>(dst),
-      TxSize::TX_16X16 => self.predict_intra_inner::<Block16x16>(dst),
-      TxSize::TX_32X32 => self.predict_intra_inner::<Block32x32>(dst),
+      TxSize::TX_4X4 => self.predict_intra_inner::<Block4x4>(dst, bit_depth),
```

Comment:
Ideally we could const-propagate the bit_depth the same way we do with blocksize.

It can happen at a later time if you do not feel confident about it.

Reply:
I prefer to leave this for a future PR.

Reply:
It's also less critical, it doesn't control branching.